Byte array converter + converters cleanup

roimenashe opened the pull request:
1. Byte arrays should not be converted or waste time on unnecessary convert collection flow (read flow only, write flow treat Byte arrays as simple fields).
2. Aerospike converters cleanup, remove unnecessary/unused converters.

reugn reviewed (commented):
We may need to add the same check to the `MappingAerospikeWriteConverter`.

roimenashe commented:
> We may need to add the same check to the `MappingAerospikeWriteConverter`.

@reugn 
Byte[] fields are written directly (without conversions), method `MappingAerospikeWriteConverter.getValueToWrite()` checks wether the field value is a simple value and in line 128 `conversions.isSimpleType(value.getClass()))` is true.

Also test `MappingAerospikeConverterTypesTest.ObjectWithByteArrayFieldWithOneValueInData()` covers write and read of byte[] fields.

Aloren commented on src/main/java/org/springframework/data/aerospike/convert/MappingAerospikeReadConverter.java:
```diff
@@ -132,6 +132,10 @@ private <T> T readValue(Object source, TypeInformation<?> propertyType) {
 			return (T) conversionService.convert(source, targetClass);
 		} else if (propertyType.isCollectionLike()) {
+			/* Byte arrays should not be converted or waste time on unnecessary convert collection flow */
+			if (source instanceof byte[]) {
```
Please prepare a test where:
1. you write data using document:
```
class Doc {
byte[] array;
}
```
2. you read data using document:
```
class Doc {
List<Byte> array;
}
```

Existing implementation should not fail, but introduced change will fail with exception. This is because you need to check not only for source, but also for target type.

roimenashe commented:
@Aloren 
Added a check for the target type - convert as it is only in case the source is byte[] and the target type is also byte[] ("[B") otherwise convert as collection.
Also added the tests that you suggested.
